dgp/utils/render_3d_to_2d.py

```diff
@@ -0,0 +1,422 @@
+# Copyright 2023 Toyota Motor Corporation.  All rights reserved.
+import json
+import logging
+import math
+import os
+import time
+from collections import defaultdict
+from multiprocessing import Pool
+from typing import Optional, Tuple
+
+import numpy as np
+from matplotlib import pyplot as plt
+
+from dgp.annotations import ANNOTATION_TYPE_TO_ANNOTATION_GROUP
+from dgp.constants import ALL_ANNOTATION_TYPES, DATASET_SPLIT_NAME_TO_KEY
+from dgp.datasets.synchronized_dataset import (
+    SynchronizedScene,
+    SynchronizedSceneDataset,
+)
+from dgp.proto.dataset_pb2 import SceneDataset
+from dgp.utils.camera import Camera
+from dgp.utils.structures.bounding_box_3d import BoundingBox3D
+
+ANNOTATIONS_3D = 'bounding_box_3d'
+ANNOTATIONS_2D = 'bounding_box_2d'
+
+
+def render_bounding_box_3d_to_2d(bbox_3d: BoundingBox3D, camera: Camera) -> np.ndarray:
+    """Render the bounding box from 3d to 2d to get the centroid.
+    Parameters
+    ----------
+    bbox_3d: BoundingBox3D
+        3D bounding box (cuboid) that is centered at `pose` with extent `sizes.
+    camera: dgp.utils.camera.Camera
+        Camera used to render the bounding box.
+    Returns
+    ----------
+    centroid: np.ndarray
+        Centroid in image plane.
+    Raises
+    ------
+    TypeError
+        Raised if camera is not an instance of Camera.
+    """
+    if not isinstance(camera, Camera):
+        raise TypeError('`camera` should be of type Camera')
+    points2d = camera.project(bbox_3d.corners)
+    corners = points2d.T
+    if (bbox_3d.corners[:, 2] <= 0).any():
+        return None
+    # Get the centroid in image plane.
+    return camera.project(np.vstack([bbox_3d.pose.tvec, bbox_3d.pose.tvec, bbox_3d.pose.tvec])).astype(np.int32)
```

Comment:
what should this return if the box is not in view or projects behind the camera?

Reply:
if the box is not in view or projects behind the camera, it will be None.